src/node_http_parser.cc

```diff
@@ -499,6 +501,11 @@ class Parser : public AsyncWrap, public StreamListener {
       max_http_header_size = env->options()->max_http_header_size;
     }
 
+    if (args.Length() > 3) {
+      CHECK(args[3]->IsBoolean());
+      lenient = args[3]->IsTrue();
+    }
```

Comment:
given that the default would be `false`, I would just simplify this to

```c++
lenient = args[3]->IsTrue();
```

Without the `if` and `CHECK` guards.

Reply:
Indexing past the end of an array is undefined, AFAIK, and the behaviour of `IsTrue()` when called on a non-Boolean is undocumented (though possibly safe), I'd rather be explicit here.

Reply:
I’m fine with either, but just for clarification:

> Indexing past the end of an array is undefined, AFAIK

Not sure if you’re referring to undefined behaviour or the `undefined` value, but yes, `args[n]` is the JS `undefined` value if `n` is out of bounds. I’ll submit a CL to V8 to clarify this.

> and the behaviour of `IsTrue()` when called on a non-Boolean is undocumented (though possibly safe), I'd rather be explicit here.

You can call `IsTrue()` on any JS value. It’s not undocumented, and I’m not sure how to improve the docs, but I’ll try my best.

Reply:
https://chromium-review.googlesource.com/c/v8/v8/+/1954386

Reply:
I simplified the code. `grep -A1 'CHECK.*args.Length' src/*.cc` has page after page of checks against length guarding checks against or uses of values as types, I had the strong impression this was the standard way we did things.

Reply:
For *most* types of arguments, I'd say checking `args.Length()` is perfectly fine. In this case, since the default is `false` and we're explicitly looking for `IsTrue()`, it just simplifies things a bit.